src/main/java/com/google/api/codegen/go/GoGapicContext.java

```diff
@@ -494,19 +500,22 @@ private GoImport createMessageImport(MessageType messageType) {
    * indentation within the 'import' section in Go file.
    */
   public Iterable<String> getImports(Interface service) {
+    Set<GoImport> standard = getStandardImports(service);
     TreeSet<GoImport> thirdParty = new TreeSet<>();
 
     // Add non-service-specific imports.
     thirdParty.add(GoImport.create("golang.org/x/net/context"));
     thirdParty.add(GoImport.create("google.golang.org/grpc"));
-    thirdParty.add(GoImport.create("google.golang.org/grpc/codes"));
     thirdParty.add(GoImport.create("google.golang.org/grpc/metadata"));
     thirdParty.add(GoImport.create(GAX_PACKAGE_BASE, "gax"));
     thirdParty.add(GoImport.create("google.golang.org/api/option"));
     thirdParty.add(GoImport.create("google.golang.org/api/transport"));
 
+    if (standard.contains(GoImport.create("time"))) {
+      thirdParty.add(GoImport.create("google.golang.org/grpc/codes"));
```

Comment:
This is a good point. I think the easiest solution is to inline the standard imports function. The function used to make sense separately, but with this kind of interaction, I think it's cleaner to bring the logic in one place.